File: leetcode/reverse-integer/dart/solution1/playground.dart
void main() {
  final solution = Solution();

  final testCases = [
    {'x': 123, 'expected': 321},
    {'x': -123, 'expected': -321},
    {'x': 120, 'expected': 21},
    {'x': 0, 'expected': 0},
    {'x': 1000, 'expected': 1},
    {'x': 1001, 'expected': 1001},
    {
      'x': 1534236469, // positive overflow
      'expected': 0,
    },
    {
      'x': -2147483412, // valid within 32-bit range
      'expected': -2143847412,
    },
    {
      'x': -1563847412, // negative overflow when reversed
      'expected': 0,
    },
  ];

  for (var i = 0; i < testCases.length; i++) {
    final x = testCases[i]['x'] as int;
    final expected = testCases[i]['expected'] as int;

    final result = solution.reverse(x);

    print('Test case ${i + 1}:');
    print('Input: x = $x');
    print('Output: $result');
    print('Expected: $expected');
    print(result == expected ? '✅ Passed\n' : '❌ Failed\n');
  }
}

File: leetcode/reverse-integer/kotlin/solution2/playground.kt
fun main() {
    val solution = Solution()

    val testCases = listOf(
        mapOf("x" to -10, "expected" to -1),
        mapOf("x" to 10, "expected" to 1),
        mapOf("x" to 123, "expected" to 321),
        mapOf("x" to -123, "expected" to -321),
        mapOf("x" to 120, "expected" to 21),
        mapOf("x" to 0, "expected" to 0),
        mapOf("x" to 1534236469, "expected" to 0), // overflow case
        mapOf("x" to -2147483412, "expected" to -2143847412)
    )

    for ((index, test) in testCases.withIndex()) {
        val x = test["x"] as Int
        val expected = test["expected"] as Int

        val result = solution.reverse(x)

        println("Test case ${index + 1}:")
        println("Input: x = $x")
        println("Output: $result")
        println("Expected: $expected")
        if (result == expected) println("✅ Passed\n") else println("❌ Failed\n")
    }
}
